core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -195,6 +205,109 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
     throw new UnsupportedOperationException("Cannot rename Hadoop tables");
   }
 
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> meta) {
+    Preconditions.checkArgument(!namespace.isEmpty(),
+        "Cannot create namespace with invalid name: %s", namespace.toString());
+    Preconditions.checkArgument(meta == null || meta.size() == 0,
+        "Hadoop Catalog not support namespace metaData: %s", namespace.toString());
+
+    Path nsPath = new Path(SLASH.join(warehouseLocation, SLASH.join(namespace.levels())));
+    FileSystem fs = Util.getFs(nsPath, conf);
+    try {
+      fs.mkdirs(nsPath);
+    } catch (IOException e) {
+      throw new RuntimeIOException("Create namespace failed: %s", namespace.toString(), e.getMessage());
+    }
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) {
+ //   Preconditions.checkArgument(!namespace.isEmpty(), "Your Namespace must be not empty!");
+    List<Namespace> namespaceList = new ArrayList<>();
+    String[] namespaces;
+    Path nsPath = new Path(SLASH.join(warehouseLocation, SLASH.join(namespace.levels())));
+    FileSystem fs = Util.getFs(nsPath, conf);
+    try {
+      if (!fs.exists(nsPath) || !fs.isDirectory(nsPath)) {
+        throw new NoSuchNamespaceException("namespace does not exist: " + namespace.toString());
+      }
+
+      List<String> pathList =  Stream.of(fs.listStatus(nsPath)).map(FileStatus::getPath)
+          .collect(Collectors.toList()).stream().filter(path -> isNamespace(fs, path))
+          .map(Path::getName).collect(Collectors.toList());
+
+      for (String path : pathList) {
+        if (!namespace.isEmpty()) {
+          namespaces = Namespace.of(namespace.toString() + "." + path).levels();
+        } else {
+          namespaces = new String[]{path};
+        }
+        namespaceList.add(Namespace.of(namespaces));
+      }
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to list namespace under: " + namespace.toString(), ioe);
+    }
+
+    return namespaceList;
+  }
+
+  @Override
+  public boolean dropNamespace(Namespace namespace, boolean cascade) {
+    if (!cascade) {
+      Preconditions.checkArgument(!namespace.isEmpty(),
+          "namespace does not exist: %s", namespace.toString());
+      Preconditions.checkArgument(listTables(namespace).size() == 0,
+          "This Namespace have tables, cannot drop it");
+      Preconditions.checkArgument(listNamespaces(namespace).size() == 0,
+          "This Namespace have sub Namespace, cannot drop it");
+    }
+    Path nsPath = new Path(SLASH.join(warehouseLocation, SLASH.join(namespace.levels())));
+    FileSystem fs = Util.getFs(nsPath, conf);
+
+    try {
+      return fs.delete(nsPath, true);
```

Comment:
When passing a boolean argument, please add a comment to signal intent. This is needed because Java doesn't accept named arguments.

```java
      return fs.delete(nsPath, true /* recursive */);
```